sdk/core/Azure.Core/src/Pipeline/HttpWebRequestTransport.cs

```diff
@@ -312,6 +312,19 @@ public HttpWebRequestImplementation()
             private string? _clientRequestId;
             private readonly Dictionary<string, List<string>> _headers = new Dictionary<string, List<string>>(StringComparer.OrdinalIgnoreCase);
 
+            protected internal override void SetHeader(string name, string value)
+            {
+                if (!_headers.TryGetValue(name, out List<string> values))
```

Comment:
Should we optimize and store a single value without the list?

Reply:
We could do that - we'd just need to handle the cases where we upgrade from a single value to a list when `Add` is called.